packages/devtools_app/lib/src/screens/memory/panes/diff/controller/class_data.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import '../../../../../shared/memory/adapted_heap_data.dart';
+import '../../../shared/heap/class_filter.dart';
+import '../../../shared/heap/heap.dart';
+import '../../../shared/primitives/simple_elements.dart';
+import 'heap_diff.dart';
+
+class RetainingPathController {
```

Comment:
nit: rename to RetainingPathData

Reply:
I think this one is controller, because it really controls how we show retaining path. User choose what details they want and we rebuild the widget.